src/random.cpp

```diff
@@ -140,6 +145,14 @@ static void manage_random (bool init)
             randombytes_close ();
         }
     }
+
+#elif defined(ZMQ_USE_LIBSODIUM)
+    if (init) {
+        int rc = sodium_init ();
```

Comment:
Since this does not use the refcount, this means that sodium_init and randombytes_close will be called for each context. Is that what you meant?

Reply:
Yes - that is how it worked before my initial change. They were called inline in the Context class constructor and destructor.

In reality I think randombytes_close is mostly a no-op in libsodium, and nobody ever reported a problem with the multiple sodium_init calls either (with libsodium again).

Reply:
Ok, just wanted to be sure.

Reply:
Adding a clarifying correction for anyone who comes here in the future: this is only a no-op if getrandom is unavailable (already excluded in the above condition). If not, libsodium has all the same threadsafety issues as tweetnacl, so this PR causes crashes because libsodium has all the same requirements as tweetnacl here.